### PR TITLE
Add getLocationBounds callback #218

### PR DESCRIPTION
**Pull request by brendanheywood**

### CODE REVIEW

File: src/L.Control.Locate.js [hunk 1]
```diff
@@ -168,6 +168,19 @@ You can find the project at: https://github.com/domoritz/leaflet-locatecontrol
...
+             * This callback can be used to override the viewport tracking
+             * This function should return a LatLngBounds object.
+             *
+             * For example to extend the viewport to ensure that a particular LatLng is visible:
+             *
+             * getLocationBounds: function(locationEvent, locateControl) {
+             *    return locationEvent.bounds.extend([-33.873085, 151.219273]);
+             * },
+             */
+            getLocationBounds: function (locationEvent, locateControl) {
```
Comment (domoritz) [inline src/L.Control.Locate.js]: isn't `locateControl` just `this` in the function context?

Comment (brendanheywood) [reply src/L.Control.Locate.js]: Yep fixed this up now thanks

**Comment by domoritz** [comment]
Don't you need to now bind this so that one can access the locate control in the callback?

**Comment by brendanheywood** [comment]
No leaflet core handles this for free, `this` is the control

**Comment by domoritz** [comment]
Sweet. I feel at some point I want to move all functions in the options to be just functions of the locate control itself. You could still patch them but they wouldn't be in the options anymore.

**Pull request merged**